### PR TITLE
Mount options for OEM and state in config

### PR DESCRIPTION
**Pull request by tetatetit**
rancher.state.oem_opt and rancher.state.opt appropriately. Especially useful common `discard`, `noatime` and BTRFS `ssd`, `subvol`

### CODE REVIEW

File: util/util_linux.go [hunk 1]
```diff
@@ -25,7 +26,14 @@ func mountProc() error {
 	return nil
 }
 
-func Mount(device, directory, fsType, options string) error {
+func Mount(device, directory, fsType string, options_i interface{}) error {
```
Comment (joshwget) [inline util/util_linux.go]: Can you use `options ...string` instead of `interface{}` here?

**Comment by joshwget** [comment]
I can definitely see this PR being useful! I think using `interface{}` is the only thing I'd change.

File: util/util_linux.go (hunk 1 above)
Comment (tetatetit) [inline util/util_linux.go]: I used interface purposely to accept both slices and string in this func so but if you think it is excess in this code - I'll get back to string only, no problem

Comment (joshwget) [inline util/util_linux.go]: You should be able to do the same with `...string`. `interface{}` is a little overkill here since it'll accept any type.

**Comment by SvenDowideit** [comment]
I'd like to see this PR contain a little user facing docs - but yeah, me too.

**Comment by tetatetit** [comment]
Done! Just getting accustomed to Go after C++ :)

File: config/schema.go [hunk 2]
```diff
@@ -157,13 +157,15 @@ var schema = `{
...
         "dev": {"type": "string"},
+        "opt": {"$ref": "#/definitions/list_of_strings"},
         "wait": {"type": "boolean"},
         "required": {"type": "boolean"},
         "autoformat": {"$ref": "#/definitions/list_of_strings"},
         "mdadm_scan": {"type": "boolean"},
         "script": {"type": "string"},
         "oem_fstype": {"type": "string"},
         "oem_dev": {"type": "string"}
+        "oem_opt": {"$ref": "#/definitions/list_of_strings"},
```
Comment (joshwget) [inline config/schema.go]: It looks like this file is out of sync with what's in `schema.json`. You can run `go run scripts/inline_schema.go` to correct this.

**Comment by joshwget** [comment]
LGTM once tests are passing!

**Comment by SvenDowideit** [comment]
@tetatetit are you interested in fixing the failing test `validate_test.go:15: invalid character '"' after object key:value pair` and adding docs? this has been waiting for a while.

**Pull request closed without merging**